user_guide_src/source/libraries/publisher.rst

```diff
@@ -0,0 +1,440 @@
+	$frameworkPublisher->addFiles([
+		'/opt/mail/susan',
+		'/opt/mail/ubuntu',
+	]);
+	$frameworkPublisher->addDirectory(SUPPORTPATH . 'Images');
+
+Once all the files are staged use one of the output commands (**copy()** or **merge()**) to process the staged files
+to their destination(s)::
+
+	// Place all files into $destination
+	$frameworkPublisher->copy();
+
+	// Place all files into $destination, overwriting existing files
+	$frameworkPublisher->copy(true);
+
+	// Place files into their relative $destination directories, overwriting and saving the boolean result
+	$result = $frameworkPublisher->merge(true);
+
+See the Library Reference for a full description of available methods.
+
+Automation and Discovery
+========================
+
+You may have regular publication tasks embedded as part of your application deployment or upkeep. ``Publisher`` leverages
+the powerful ``Autoloader`` to locate any child classes primed for publication::
+
+	use CodeIgniter\CLI\CLI;
+	use CodeIgniter\Publisher\Publisher;
+	
+	foreach (Publisher::discover() as $publisher)
+	{
+		$result = $publisher->publish();
+
+		if ($result === false)
+		{
+			CLI::write(get_class($publisher) . ' failed to publish!', 'red');
+		}
+	}
+
+By default ``discover()`` will search for the "Publishers" directory across all namespaces, but you may specify a
+different directory and it will return any child classes found::
+
+	$memePublishers = Publisher::discover('CatGIFs');
+
+Most of the time you will not need to handle your own discovery, just use the provided "publish" command::
+
+	> php spark publish
+
+By default on your class extension ``publish()`` will add all files from your ``$source`` and merge them
+out to your destination, overwriting on collision.
+
+********
+Examples
+********
+
+Here are a handful of example use cases and their implementations to help you get started publishing.
+
+File Sync Example
+=================
+
+You want to display a "photo of the day" image on your homepage. You have a feed for daily photos but you
+need to get the actual file into a browsable location in your project at **public/images/daily_photo.jpg**.
+You can set up :doc:`Custom Command </cli/cli_commands>` to run daily that will handle this for you::
+
+	<?php
+
+	namespace App\Commands;
+
+	use CodeIgniter\CLI\BaseCommand;
+	use CodeIgniter\Publisher\Publisher;
+	use Throwable;
+
+	class DailyPhoto extends BaseCommand
+	{
+		protected $group       = 'Publication';
+		protected $name        = 'publish:daily';
+		protected $description = 'Publishes the latest daily photo to the homepage.';
+
+		public function run(array $params)
+		{
+			$publisher = new Publisher('/path/to/photos/', FCPATH . 'assets/images');
+
+			try
+			{
+				$publisher->addPath('daily_photo.jpg')->copy($replace = true);
+			}
+			catch (Throwable $e)
+			{
+				$this->showError($e);
+			}
+		}
+	}
+
+Now running ``spark publish:daily`` will keep your homepage's image up-to-date. What if the photo is
```

Comment:
I do not see a `publish:daily` command. Is this still to be made?

Reply:
It's the example command immediately above 🙃